cdap-app-templates/cdap-etl/cdap-etl-realtime/src/main/java/co/cask/cdap/templates/etl/realtime/ETLRealtimeTemplate.java

```diff
@@ -0,0 +1,142 @@
+import co.cask.cdap.templates.etl.common.DefaultStageConfigurer;
+import co.cask.cdap.templates.etl.common.ETLTemplate;
+import co.cask.cdap.templates.etl.realtime.config.ETLRealtimeConfig;
+import co.cask.cdap.templates.etl.realtime.sinks.NoOpSink;
+import co.cask.cdap.templates.etl.realtime.sources.TestSource;
+import co.cask.cdap.templates.etl.transforms.IdentityTransform;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.gson.Gson;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * ETL Realtime Template.
+ */
+public class ETLRealtimeTemplate extends ETLTemplate<ETLRealtimeConfig> {
+  public static final String STATE_TABLE = "etlrealtimesourcestate";
+  private static final Gson GSON = new Gson();
+  private final Map<String, String> sourceClassMap;
+  private final Map<String, String> sinkClassMap;
+  private final Map<String, String> transformClassMap;
+  private RealtimeSource source;
+  private RealtimeSink sink;
+  private List<Transform> transforms;
+
+  public ETLRealtimeTemplate() throws Exception {
+    sourceClassMap = Maps.newHashMap();
+    sinkClassMap = Maps.newHashMap();
+    transformClassMap = Maps.newHashMap();
+    transforms = Lists.newArrayList();
+
+    // Add class from lib here to be made available for use in the ETL Worker.
+    // TODO : Remove this when plugins management is available.
+    initTable(Lists.<Class>newArrayList(IdentityTransform.class,
+                                        NoOpSink.class,
+                                        TestSource.class));
+  }
+
+  private void initTable(List<Class> classList) throws Exception {
+    for (Class klass : classList) {
+      DefaultStageConfigurer configurer = new DefaultStageConfigurer(klass);
+      if (RealtimeSource.class.isAssignableFrom(klass)) {
+        RealtimeSource source = (RealtimeSource) klass.newInstance();
+        source.configure(configurer);
+        sourceClassMap.put(configurer.createSpecification().getName(), configurer.createSpecification().getClassName());
+      } else if (RealtimeSink.class.isAssignableFrom(klass)) {
+        RealtimeSink sink = (RealtimeSink) klass.newInstance();
+        sink.configure(configurer);
+        sinkClassMap.put(configurer.createSpecification().getName(), configurer.createSpecification().getClassName());
+      } else {
+        Preconditions.checkArgument(Transform.class.isAssignableFrom(klass));
+        Transform transform = (Transform) klass.newInstance();
+        transform.configure(configurer);
+        transformClassMap.put(configurer.createSpecification().getName(),
+                              configurer.createSpecification().getClassName());
+      }
+    }
+  }
+
+  @Override
+  public void configureAdapter(String adapterName, ETLRealtimeConfig etlConfig, AdapterConfigurer configurer)
+    throws Exception {
+    ETLStage sourceConfig = etlConfig.getSource();
```

Comment:
most of this is still common between realtime and batch. can have a configureAdapter in parent, then children call super() and only do their specific parts, setting instances or setting a schedule.